api/src/database/db.ts

```diff
@@ -29,10 +29,19 @@ const poolConfig: PoolConfig = {
 
 defaultLog.debug({ label: 'create db pool', message: 'pool config', poolConfig });
 
-let pool: Pool;
+// Custom type handler for psq `DATE` type to prevent local time/zone information from being added.
+// Why? By default, node-postgres assumes local time/zone for any psql `DATE` or `TIME` types that don't have timezone information.
+// This Can lead to unexpected behaviour when the original psql `DATE` value was intentionally omitting time/zone information.
+// PSQL date types: https://www.postgresql.org/docs/12/datatype-datetime.html
+// node-postgres type handling (see bottom of page): https://node-postgres.com/features/types
+pg.types.setTypeParser(1082, (stringValue: string) => {
```

Comment:
```suggestion
pg.types.setTypeParser(pg.types.builtins.DATE, (stringValue: string) => {
```

Reply:
Do we need to set a similar typeparser for timestamp and timestampz?

Reply:
oh nice one! I didn't realize there was a built in variable for that.

Right now its only `DATE` that we have issues with.  `timestamp(z)` already includes time/timezone information, so there is no issues as node-postgres doesn't alter that in any way.  Its only an issue for the types that don't have time/timezone information, as node-postgres decides to add it anyways, using some default values.  

I didn't add a function for `TIME`, which is the other type that node-postgres messes with, as we dont have any UI that creates time values yet.  We really only have 4 date fields in the whole app: start/end date for project and survey.